src/GraphQL/Types/Relay/PageInfoType.cs

```diff
@@ -13,8 +13,8 @@ public PageInfoType()
 
             Field<NonNullGraphType<BooleanGraphType>>("hasNextPage", "When paginating forwards, are there more items?");
             Field<NonNullGraphType<BooleanGraphType>>("hasPreviousPage", "When paginating backwards, are there more items?");
-            Field<StringGraphType>("startCursor", "When paginating backwards, the cursor to continue.");
-            Field<StringGraphType>("endCursor", "When paginating forwards, the cursor to continue.");
+            Field<NonNullGraphType<StringGraphType>>("startCursor", "When paginating backwards, the cursor to continue.");
+            Field<NonNullGraphType<StringGraphType>>("endCursor", "When paginating forwards, the cursor to continue.");
```

Comment:
consider me educated. i have reverted this

Reply:
> I don't think we can make these non-nullable. If there are no results, these would be null. See the reference specification schema. https://relay.dev/docs/guides/graphql-server-specification/

There seems to be conflicting advice in the documentation. `startCursor` and `endCursor` are marked as nullable in [the schema in the link you provided @Shane32 ](https://relay.dev/docs/guides/graphql-server-specification/#schema), but following the link to the [GraphQL Cursor Connections spec](https://relay.dev/assets/files/connections-932f4f2cdffd79724ac76373deb30dc8.htm#sel-EALFDDAAACJvB_iO) suggests the opposite. Namely:

> PageInfo must contain fields hasPreviousPage and hasNextPage, both of which return non-null booleans. It must also contain fields startCursor and endCursor, both of which return non-null opaque strings.

The [Introspection of PageInfo](https://relay.dev/assets/files/connections-932f4f2cdffd79724ac76373deb30dc8.htm#sec-undefined.PageInfo.Introspection) in this doco lists all `PageInfo` properties as `"kind": "NON_NULL"`

How do we tell which one is correct?

Reply:
Even within the spec it is confusing:
> An “Edge Type” must contain a field called cursor. This field must return a type that serializes as a String; this may be a String, a Non-Null wrapper around a String, a custom scalar that serializes as a String, or a Non-Null wrapper around a custom scalar that serializes as a String.
>
> Whatever type this field returns will be referred to as the cursor type in the rest of this spec.

So there it says that cursor fields may be nullable.  But down below as you say, it explicitly says a "non-null string" and does not mention the words "cursor type" as quoted above.

I think that common sense must prevail here and dictate that when there are no records returned, there surely cannot be a starting cursor and ending cursor.  I know that if the cursor was an index, then one could use `int.MinValue` and `int.MaxValue` -- but it is not.  For example, if the cursor on an item is an ID field of type Guid, there is no possible guid that could represent the starting cursor or ending cursor for a connection that returned no results.  So I think it is clear that these fields must be nullable.

If we assume that startCursor and endCursor must be nullable, then we have two options before us:

1. Use common sense and specify the cursor field's type of an edge type to be a non-null string value.
2. Match the cursor field's type of the edge type to be nullable to match the startCursor and endCursor field types.

Let us also consider one final thing: how does typical executing code use these fields?  Well, it would store the startCursor or endCursor fields verbatim within a variable until needed to pass to the 'after' or 'before' property of another connection request.  No parsing or any other type of processing should take place.  So in this instance, 'after' or 'before' can be nullable, and indeed would always return the first page of a list if a user tried to navigate forward or backward from a connection that returned no results with null startCursor and endCursor fields.  This seems to be perfectly logical behavior.

I suggest implementing the changes shown here, where the edge type's cursor field is non-nullable, but the pageinfo's startCursor and endCursor fields are nullable.